### PR TITLE
Use yaml

### PR DESCRIPTION
**Pull request by hlieberman**
Though this change is primarily driven by my need to be able to specify a collection in the properties file, this also reduces the amount of custom code that we have to carry around.

### CODE REVIEW

File: shared_python/Args.py [hunk 1]
```diff
@@ -11,15 +12,11 @@ def _load_args_from_file(filepath):
   props = {}
```
Comment (ariana-paris) [inline shared_python/Args.py]: Looks as if we don't need `props` or `sep` anymore so these can be removed

File: shared_python/Args.py [hunk 2]
```diff
@@ -11,15 +12,11 @@ def _load_args_from_file(filepath):
...
-      l = line.strip()
-      if l:
-        key_value = l.split(sep)
-        key = key_value[0].strip()
-        value = sep.join(key_value[1:]).strip().strip('"')
-        props[key] = value
-  return props
-
+    try:
+      x = yaml.safe_load(f)
```
Comment (ariana-paris) [inline shared_python/Args.py]: Is there a reason we can't return the result of `yaml.safe_load(f)` directly as `x` doesn't add much to the legibility here?

File: shared_python/Args.py [hunk 3]
```diff
@@ -11,15 +12,11 @@ def _load_args_from_file(filepath):
...
+      return x
+    except yaml.YAMLError as exc:
+      log.error("YAML parsing failure: {}".format(exc))
```
Comment (ariana-paris) [inline shared_python/Args.py]: Is this just swallowing the error and returning a `None` that will leap out and surprise us later? Or does it just bomb out at this point (which is probably preferable)?

Comment (hlieberman) [reply shared_python/Args.py]: Yes, that was the idea, as that was what we were doing before.  (Swallowing the error).  I agree that throwing an exception is better practice, though, and if you're willing to let us go that way, we should!

**Comment by hlieberman** [comment]
@ariana-paris corrected those requests.  I'm not sure why the rebase was needed -- it should have been a straight ff from master, since the ancestor commits were in there already.  But, it was easily done!

**Pull request merged**